compiler/ir/backend.common/src/org/jetbrains/kotlin/backend/common/lower/ReturnableBlockLowering.kt

```diff
@@ -69,40 +69,25 @@ import org.jetbrains.kotlin.ir.visitors.IrElementTransformer
  * }
  *
  */
-class ReturnableBlockLowering(val context: CommonBackendContext) : FileLoweringPass {
-    override fun lower(irFile: IrFile) {
-        irFile.transform(ReturnableBlockTransformer(context), ReturnableBlockLoweringContext(irFile))
-    }
-}
-
-private class ReturnableBlockLoweringContext(val containingDeclaration: IrDeclarationParent) {
-    var labelCnt = 0
-    val returnMap = mutableMapOf<IrReturnableBlockSymbol, (IrReturn) -> IrExpression>()
-}
-
-private class ReturnableBlockTransformer(
-    val context: CommonBackendContext
-) : IrElementTransformer<ReturnableBlockLoweringContext> {
+class ReturnableBlockLowering(val context: CommonBackendContext) : FileLoweringPass, IrElementTransformerVoidWithContext() {
+    private var labelCnt = 0
```

Comment:
Counter was file specific but now it's  module specific.

Reply:
Thanks, haven't noticed that the JS backend only constructs file lowerings once, unlike the JVM one.